pynamodb/attributes.py

```diff
@@ -668,16 +667,25 @@ def deserialize(self, value):
         """
         Takes a UTC datetime string and returns a datetime object
         """
+        return self._fast_parse_utc_date_string(value)
+
+    @staticmethod
+    def _fast_parse_utc_date_string(date_string):
```

Comment:
This has been moved inside the class and slightly modified to:
- check if strings have the proper length (avoids IndexError and a varying like microsecond field)
- rely on the microsecond field being left padded with 0s to remove the multiplication and rounding

Reply:
```suggestion
    def _fast_parse_utc_date_string(date_string: str) -> datetime:
```